### DISCUSSION TITLE
Sharing state between multiple non-nested recoil roots

### DISCUSSION THREAD

**Discussion by StebanDev** [facebookexperimental/Recoil#1848] [Q&A]
I wonder if it's possible to share state between multiple recoil roots not nested but at the same level.
I have tried setting the `override` prop to false but I still get a different value when trying to access the state from the atoms.
If you wonder why I just don't use a parent recoil root and have the other roots nested inside is because I am exporting(registering) these two components as separate web components but need to use a single store cause they are kind of dependent on the state of each other.

I would appreciate if someone can help me out since I'm not an expert in recoil but would like to know if it is feasible.

**Comment by drarmstr** [comment]
You might be able to use [`useRecoilBridgeAcrossReactRoots()`](https://recoiljs.org/docs/api-reference/core/useRecoilBridgeAcrossReactRoots) to help here, though it may not since that still requires the bridge to be created in the context of the host root.

If you really need to share state and there is no common component for establishing a shared RecoiRoot, then you may want to consider [Atom Effects](https://recoiljs.org/docs/guides/atom-effects) for syncing atom state.  There is also [`useRecoilSnapshot()`](https://recoiljs.org/docs/api-reference/core/useRecoilSnapshot), but be careful with that as it updates for any and all state changes.

**Reply by StebanDev** [reply]
I'm not following, I have read the docs but haven't been able to figure out how to use those APIs to get something like [this library](https://www.npmjs.com/package/react-hooks-global-state) does.

I would really appreciate if you can help me out with some examples 🙂

**Reply by Gutyn** [reply]
Any luck? WIX/React-native-navigation is unusable with recoil currently because of the root being recreated.
